security/s2a/internal/crypter/halfconn.go

```diff
@@ -0,0 +1,49 @@
+package crypter
+
+import (
+	"fmt"
+	"golang.org/x/crypto/cryptobyte"
+	"hash"
+)
+
+type S2AHalfConnection struct {
+	aeadCrypter s2aAeadCrypter
+	expander    hkdfExpander
```

Comment:
The half connection will not need to own an expander. It will just create one when it is needed (which will only be at creation time and when a key update occurs).

Reply:
Do you think it may be worth keeping as a member still? I'm assuming the half connection may have its key updated several times, in which case we would create a new expander every time. If we keep it as a member, I believe we can just reuse the same expander instead of creating one every time.

Reply:
If you want to keep it, that's totally fine so long as `expander` doesn't hold any state. If that's the case, then we might as well keep it around.

Reply:
The struct is empty `type defaultHKDFExpander struct{}` so it doesn't hold any state at all. I think it also might be worth keeping because I think the code would look better if we call `hc.expander.expand(...)` in various places rather than `newDefaultHKDFExpander().expand(...)`.

If we have to call `newDefaultHKDFExpander().expand(...)` everywhere, I think that would defeat the purpose of creating an interface for the expansion function to begin with since we would have to swap all of that to `new<Different_Library>HKDFExpander().expand(...)` in multiple places.